source/acquaman/BioXAS/BioXASXASScanActionController.cpp

```diff
@@ -73,13 +73,21 @@ AMAction3* BioXASXASScanActionController::createInitializationActions()
 	CLSSIS3820Scaler *scaler = CLSBeamline::clsBeamline()->scaler();
 
 	if (scaler) {
-//		double regionTime = double(bioXASConfiguration_->scanAxisAt(0)->regionAt(0)->regionTime());
 
-//		scalerInitialization = new AMSequentialListAction3(new AMSequentialListActionInfo3("BioXAS Scaler Initialization Actions", "BioXAS Scaler Initialization Actions"));
-//		scalerInitialization->addSubAction(scaler->createContinuousEnableAction3(false));
-//		scalerInitialization->addSubAction(scaler->createDwellTimeAction3(regionTime));
-//		scalerInitialization->addSubAction(scaler->createStartAction3(true));
-//		scalerInitialization->addSubAction(scaler->createWaitForDwellFinishedAction(regionTime + 5.0));
+		scalerInitialization = new AMSequentialListAction3(new AMSequentialListActionInfo3("BioXAS Scaler Initialization Actions", "BioXAS Scaler Initialization Actions"));
+
+		// Check that the scaler is in single shot mode and is not acquiring.
+
+		scalerInitialization->addSubAction(scaler->createContinuousEnableAction3(false));
+		scalerInitialization->addSubAction(scaler->createStartAction3(false));
```

Comment:
I think it's safest to change mode first, here, because the scaler may get into a funny state otherwise. If it isn't acquiring, the scaler isn't in Continuous mode; if the total number of scans and the scans per buffer haven't been set, it isn't technically in SingleShot either. The risk now is that the scan will fail completely.